### DISCUSSION TITLE
Docker image?

### DISCUSSION THREAD

**Discussion by RetroButton** [kagemomiji/airsonic-advanced#167] [General]
I´ve stumbled over Your repo for Airsonic Advanced, because i was looking for a active developed or maintained version of Airsonic Advanced.
Simple question: Will there be, or is there a working docker image?
Acutally i´m using a years old, but perfectly working release of Airsonic on Docker. Never had luck getting Airsonic Advanced running stable.

**Comment by bartmans** [comment]
Docker images are available at ghrc.io:

ghcr.io/kagemomiji/airsonic-advanced:latest
image: ghcr.io/kagemomiji/airsonic-advanced:stable

**Usage**

Airsonic-Advanced v10.6.x series (and its snapshots) are intercompatible with vanilla Airsonic 10.6.x series. This may not necessarily be the case with 11.x versions.

Also note that Airsonic-Advanced 11.x (and its snapshots) are breaking (non-backwards-compatible) version changes. You will not be able to revert back to 10.6.x after upgrading (the system does create a backup of the DB in case such revert is necessary, but it must be manually restored).

Airsonic-Advanced snapshots are generally pretty stable and recommended for use over the stable releases (which may be extremely outdated).

**Docker**

[GHCR](https://ghcr.io/kagemomiji/airsonic-advanced). Docker releases are recently multiplatform, which means ARMv7 and ARM64 are also released to Dockerhub. However, automated testing for those archs is not currently done in the CI/CD pipeline (only Linux platform is tested).

Please note that for Docker images, the volume mounting points have changed and are different from Airsonic. Airsonic mount points are at /airsonic/* inside the container. Airsonic-Advanced tries to use the same volume locations as the default war image at /var/* in order to remain consistent if people want to switch between the containers and non-containers.

    Music:/airsonic/music -> Music:/var/music
    Podcasts:/airsonic/podcast -> Podcasts:/var/podcast
    Playlists:/airsonic/playlists -> Playlists:/var/playlists
    /airsonic/data -> /var/airsonic

Also note that the Docker image will by default run as user root (0), group root (0), and so any files created in the external volume will be owned as such. You may change the user running the internal process in one of two ways:

    Specifying --user when invoking the docker run command, and providing it with one or both in the format uid:gid
    Specifying the PUID or PGID environment variables to the container image when invoking the docker run command (-e PUID=uid -e PGID=gid)
